pkg/xds/sync/dataplane_watchdog.go

```diff
@@ -165,7 +165,25 @@ func (d *DataplaneWatchdog) syncIngress(ctx context.Context, metadata *core_xds.
 		ControlPlane: d.EnvoyCpCtx,
 		Mesh:         xds_context.MeshContext{}, // ZoneIngress does not have a mesh!
 	}
-	proxy, err := d.IngressProxyBuilder.Build(ctx, d.key)
+
+	meshContexts, err := xds_context.AggregateMeshContexts(ctx, d.ResManager, d.MeshCache.GetMeshContext)
```

Comment:
Should we use `d.MeshCache.GetMeshContext` to get individual MeshContexts for the aggregated?

Reply:
what's the alternative?